testnet/stacks-node/src/tests/signer.rs

```diff
@@ -486,63 +720,68 @@ fn stackerdb_dkg_sign() {
             merkle_root: None,
         },
     };
+    let coordinator_sender = signer_test.get_coordinator_sender(reward_cycle);
+    let sign_now = Instant::now();
     coordinator_sender
         .send(sign_command)
         .expect("failed to send Sign command");
     coordinator_sender
         .send(sign_taproot_command)
         .expect("failed to send Sign taproot command");
-    for recv in signer_test.result_receivers.iter() {
-        let mut frost_signature = None;
-        let mut schnorr_proof = None;
-        loop {
-            let results = recv
-                .recv_timeout(timeout)
-                .expect("failed to recv signature results");
-            for result in results {
-                match result {
-                    OperationResult::Sign(sig) => {
-                        info!("Received Signature ({},{})", &sig.R, &sig.z);
-                        frost_signature = Some(sig);
-                    }
-                    OperationResult::SignTaproot(proof) => {
-                        info!("Received SchnorrProof ({},{})", &proof.r, &proof.s);
-                        schnorr_proof = Some(proof);
-                    }
-                    OperationResult::DkgError(dkg_error) => {
-                        panic!("Received DkgError {:?}", dkg_error);
-                    }
-                    OperationResult::SignError(sign_error) => {
-                        panic!("Received SignError {}", sign_error);
-                    }
-                    OperationResult::Dkg(point) => {
-                        panic!("Received aggregate_group_key {point}");
-                    }
-                }
-            }
-            if (frost_signature.is_some() && schnorr_proof.is_some())
-                || sign_now.elapsed() > timeout
-            {
-                break;
-            }
-        }
-        let frost_signature =
-            frost_signature.expect(&format!("Failed to get frost signature within {timeout:?}"));
-        assert!(
-            frost_signature.verify(&key, msg.as_slice()),
-            "Signature verification failed"
-        );
-        let schnorr_proof = schnorr_proof.expect(&format!(
-            "Failed to get schnorr proof signature within {timeout:?}"
-        ));
-        let tweaked_key = wsts::compute::tweaked_public_key(&key, None);
+    let frost_signatures = signer_test.wait_for_frost_signatures(timeout);
+    let schnorr_proofs = signer_test.wait_for_taproot_signatures(timeout);
+
+    for frost_signature in frost_signatures {
+        assert!(frost_signature.verify(&new_key, &msg));
+    }
+    for schnorr_proof in schnorr_proofs {
+        let tweaked_key = tweaked_public_key(&new_key, None);
         assert!(
-            schnorr_proof.verify(&tweaked_key.x(), &msg.as_slice()),
+            schnorr_proof.verify(&tweaked_key.x(), &msg),
```

Comment:
It would fail only if it was some bug in the code that results in different party shares contributing to the signature. If we do not meet the threshold, we don't even produce a signature.